app/src/main/java/com/example/erabook/fragments/discoverGoogle/DiscoverDetailsFragment.kt

```diff
@@ -84,28 +84,12 @@ class DiscoverDetailsFragment : Fragment() {
                 )
             }
             shareBook.setOnClickListener {
-                val bookDetailsIntent = Intent(Intent.ACTION_SEND).apply {
-                    type = "text/plain"
-                    putExtra(
-                        Intent.EXTRA_TEXT, getString(
-                            R.string.share_book,
-                            bookItem?.volumeInfo?.title,
-                            bookItem?.volumeInfo?.authors?.get(0)
-                                ?: "Authors not found",
-                            bookItem?.volumeInfo?.pageCount.toString(),
-                            bookItem?.volumeInfo?.publishedDate
-                        )
-                    )
-                    putExtra(
-                        Intent.EXTRA_SUBJECT,
-                        getString(R.string.share_book_subject)
-                    )
-                }
-                val chooserIntent = Intent.createChooser(
-                    bookDetailsIntent,
-                    getString(R.string.send_details)
+                requireActivity().startBookDetailsIntent(
+                    bookItem?.volumeInfo?.title,
```

Comment:
You're passing nullable variables to this extension function, that's bad practice! Since you don't know if it will be null or not. Please handle nullability.